datatypes/src/operations/reproject.rs

```diff
@@ -214,14 +226,60 @@ where
             .chain(left_line)
             .collect();
 
+        let proj_outline_coordinates = projector.project_coordinates(&outline_coordinates)?;
+
+        Ok(MultiPoint::new_unchecked(proj_outline_coordinates).spatial_bounds())
+    }
+}
+
+pub trait ReprojectClipped<P: CoordinateProjection> {
+    type Out;
+    /// Reproject and clip with respect to the area of use of the projection
+    fn reproject_clipped(&self, projector: &P) -> Result<Self::Out>;
+}
+
+impl<P> ReprojectClipped<P> for BoundingBox2D
+where
+    P: CoordinateProjection,
+{
+    type Out = BoundingBox2D;
+    fn reproject_clipped(&self, projector: &P) -> Result<BoundingBox2D> {
+        const POINTS_PER_LINE: i32 = 7;
+
+        let use_bbox = projector.target_srs().area_of_use()?;
+        let bbox = self
+            .intersection(&use_bbox)
+            .ok_or(Error::BboxesDoNotIntersect)?;
+
+        let upper_line = Line::new(bbox.upper_left(), bbox.upper_right())
+            .with_additional_equi_spaced_coords(POINTS_PER_LINE);
+        let right_line = Line::new(bbox.upper_right(), bbox.lower_right())
+            .with_additional_equi_spaced_coords(POINTS_PER_LINE);
+        let lower_line = Line::new(bbox.lower_right(), bbox.lower_left())
+            .with_additional_equi_spaced_coords(POINTS_PER_LINE);
+        let left_line = Line::new(bbox.lower_left(), bbox.upper_left())
+            .with_additional_equi_spaced_coords(POINTS_PER_LINE);
+
+        let outline_coordinates: Vec<Coordinate2D> = upper_line
+            .chain(right_line)
+            .chain(lower_line)
+            .chain(left_line)
+            .collect();
+
         let proj_outline_coordinates: Vec<Coordinate2D> =
             project_coordinates_fail_tolerant(&outline_coordinates, projector)
                 .into_iter()
                 .flatten()
                 .collect();
 
-        // TODO: check min coords or use grid? e.g. ensure!(proj_cs.len() ). fail only if there are not enough coordinates after reprojection. This could also require us to use a grid instead of bounds. The good news is that the Grid can already provide this.
-        Ok(MultiPoint::new_unchecked(proj_outline_coordinates).spatial_bounds())
+        let out = MultiPoint::new(proj_outline_coordinates)?.spatial_bounds();
```

Comment:
while this removes the todo it will also work with only one point

Reply:
but I check in the next line that the bbox has a real size and is not a point

Reply:
sorry did not see that -_-